### DISCUSSION TITLE
serverpod_auth_server still requires http 0.13.0

### DISCUSSION THREAD

**Discussion by gecko20** [serverpod/serverpod#1879] [Q&A] [answered]
Let me at first thank you for your work on this awesome project!

We're building an app with it and stumbled on a problem regarding the version of the http package while I attempted upgrading to Serverpod 1.2.3 from 1.1.0.

We use the web3dart package and had to upgrade it to a more recent version (>2.7) since the old one depended on uuid ^3.0.7 and Serverpod now seems to use uuid 4.3.3. After setting the version of web3dart to 2.7.2, dart pub update now prints out the following error:

```
dart pub update
Resolving dependencies... (2.1s)
Because web3dart >=2.7.0 depends on eip1559 ^0.6.2 which depends on http ^1.1.0, web3dart >=2.7.0 requires http ^1.1.0.
And because serverpod_auth_server >=0.9.10 depends on firebase_admin ^0.2.0 which depends on http ^0.13.0, web3dart >=2.7.0 is incompatible with
  serverpod_auth_server >=0.9.10.
So, because nftapp_server depends on both serverpod_auth_server 1.2.3 and web3dart 2.7.2, version solving failed.
```

Looking into serverpod_auth_server's pubspec.yaml, I saw that it indeed depends on firebase_admin ^0.2.0 which, in turn, still requires http ^0.13.0.

Is there any way to fix this or do I have to downgrade to Serverpod 1.1.0 again?

Any advice would be appreciated!

Thanks

**Comment by vlidholt** [comment]
I think http 1.x is not too different to 0.13, if you do not need 1.x you can specify it as >=0.13.0 <2.0.0 and it should work.

Unfortunately fireabase_admin isn't yet updated. In Serverpod 2, we are likely to split up the auth package into sub packages for different authentication methods. This would make it more resilient to issues like this one.

**Comment by gecko20** [comment, marked as answer]
> I think http 1.x is not too different to 0.13, if you do not need 1.x you can specify it as >=0.13.0 <2.0.0 and it should work.
> 
> Unfortunately fireabase_admin isn't yet updated. In Serverpod 2, we are likely to split up the auth package into sub packages for different authentication methods. This would make it more resilient to issues like this one.

Forking web3dart and eip1559, changing the http dependency in the latter to >=0.13.0 <2.0.0 and making web3dart depend on the fork seemed to do the trick – not a nice solution but I guess this has to do it for now.

Splitting up the auth package seems like a good idea – for example, we don't use anything Firebase related.

Now I'm looking forward to using Serverpod's new database functionalities. :)

Thanks for your help!

**Reply by vlidholt** [reply]
I converted this to a discussion so that it's a bit easier to find. :) Thanks for the update!

**Comment by jorylenz1** [comment]
In your pubspec.yaml, you should be able to add a dependency_override like below.

```
dependency_overrides:
  http: 1.2.0
```
